openlibrary/components/BulkSearch/components/BulkSearchControls.vue

```diff
@@ -71,74 +70,22 @@ export default {
             }
         }
     },
+    computed: {
+        showApiKey(){
+            if (this.bulkSearchState.extractionMethod) return 'model' in this.bulkSearchState.extractionMethod
+            return false
+        }
+    },
     methods: {
         togglePasswordVisibility(){
             this.showPassword= !this.showPassword
         },
         selectAlgorithm(e) {
```

Comment:
Instead of making this a method, this can now work as a getter on `BulkSearchState`!

```
get activeExtractor() {
    return this.extractors[this.bulkSearchState._activeExtractorIndex];
}
```

This is a more common pattern in reactive/vue land. Then the select becomes `v-model="bulkSearchState._activeExtractorIndex"`.

Reply:
I like this idea too. Resolved!